src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -377,38 +378,313 @@ internal PhysicalLocationVersionOne CreatePhysicalLocation(Region v2Region)
+                {
+                    reader.Dispose();
+                }
+            }
+        }
+
+        private int GetRegionByteOffset(Region v2Region, Uri uri)
+        {
+            int result = 0;
+            TextReader reader = null;
+
+            try
+            {
+                Encoding encoding;
+                reader = GetFileTextReader(uri, out encoding);
+
+                if (reader != null)
+                {
+                    char[] buffer = new char[v2Region.CharOffset];
+
+                    // Read everything up to charOffset
+                    if (reader.ReadBlock(buffer, 0, buffer.Length) > 0)
+                    {
+                        result = encoding.GetByteCount(buffer, 0, buffer.Length);
+                    }
+                }
+            }
+            finally
+            {
+                if (reader != null)
+                {
+                    reader.Dispose();
+                }
+            }
+
+            return result;
+        }
+
+        private int GetRegionByteLength(Region v2Region, Uri uri)
+        {
+            int result = 0;
+            TextReader reader = null;
+
+            try
+            {
+                Encoding encoding;
+                reader = GetFileTextReader(uri, out encoding);
+
+                if (v2Region.StartLine > 0) // Use line and column 
+                {                    
+                    string sourceLine = null;
+
+                    // Read down to the line before startLine
+                    for (int i = 1; i < v2Region.StartLine; sourceLine = reader.ReadLine(), i++) { }
+
+                    // Read to startColumn
+                    char[] buffer = new char[v2Region.StartColumn];
+                    reader.Read(buffer, 0, buffer.Length);
+
+                    // Read the next charLength characters
+                    buffer = new char[v2Region.CharLength];
+                    reader.Read(buffer, 0, buffer.Length);
+
+                    result = encoding.GetByteCount(buffer);
+                }
+                else // Use charOffset
+                {
+                    // Read the next charLength characters
+                    char[] buffer = new char[v2Region.CharOffset];
+                    reader.Read(buffer, 0, buffer.Length);
+
+                    result = encoding.GetByteCount(buffer);
+                }
+            }
+            finally
+            {
+                if (reader != null)
+                {
+                    reader.Dispose();
+                }
+            }
+
+            return result;
+        }
+
+        private int GetRegionEndColumn(Region v2Region, Uri uri)
+        {
+            int result = 0;
+            TextReader reader = null;
+
+            try
+            {
+                Encoding encoding;
+                reader = GetFileTextReader(uri, out encoding);
+                string sourceLine = null;
+
+                // Read down to the line before endLine
```

Comment:
Bug: You need the length of the `endLine`'th line, not the one before it.

Reply:
Fixed

Reply:
Fixed